kf/kf_unix.go

```diff
@@ -115,6 +115,43 @@ func VerifyNMountBPFFS() error {
 			return fmt.Errorf("unable to mount %s at %s: %s", srcPath, dstPath, err)
 		}
 	}
+
+	return VerifyNMountTraceFS()
+}
+
+// VerifyNMounTraceFS - Mounting trace filesystem
+func VerifyNMountTraceFS() error {
+	dstPath := "/sys/kernel/tracing"
+	srcPath := "tracefs"
+	fstype := "tracefs"
+	flags := syscall.MS_NODEV | syscall.MS_NOEXEC | syscall.MS_NOSUID | syscall.MS_RELATIME
+
+	mnts, err := ioutil.ReadFile("/proc/mounts")
+	if err != nil {
+		return fmt.Errorf("failed to read procfs: %v", err)
+	}
+
+	if !strings.Contains(string(mnts), dstPath) {
```

Comment:
Logic here should be
if traceFs is not mounted at all:
   mount at location we want
else if 
traceFS _is_ mounted _but_ at incorrect location:
   umount
   remount where we want

Reply:
I would suggest we don't unmount existing as it might break something else.  I think the correct behavior is to just mount again under the path we want or bind mount to the location we need

Reply:
+1 to Jason's comment

Reply:
Agreed